tesseract_collision/core/src/types.cpp

```diff
@@ -57,57 +57,209 @@ void ContactResult::clear()
 
 ContactRequest::ContactRequest(ContactTestType type) : type(type) {}
 
-std::size_t flattenMoveResults(ContactResultMap&& m, ContactResultVector& v)
+ContactResult& ContactResultMap::addContactResult(const KeyType& key, ContactResult result)
 {
-  v.clear();
-  v.reserve(m.size());
-  for (const auto& mv : m)
+  ++cnt_;
+  auto& cv = data_[key];
+  return cv.emplace_back(std::move(result));
+}
+
+ContactResult& ContactResultMap::addContactResult(const KeyType& key, const MappedType& results)
+{
+  assert(!results.empty());
+  cnt_ += static_cast<long>(results.size());
+  auto& cv = data_[key];
+  ;
+  cv.reserve(cv.size() + results.size());
+  cv.insert(cv.end(), results.begin(), results.end());
+  return cv.back();
+}
+
+ContactResult& ContactResultMap::setContactResult(const KeyType& key, ContactResult result)
+{
+  auto& cv = data_[key];
+  cnt_ += (1 - static_cast<long>(cv.size()));
+  assert(cnt_ >= 0);
+  cv.clear();
+
+  return cv.emplace_back(std::move(result));
+}
+
+ContactResult& ContactResultMap::setContactResult(const KeyType& key, const MappedType& results)
+{
+  assert(!results.empty());
+  auto& cv = data_[key];
+  cnt_ += (static_cast<long>(results.size()) - static_cast<long>(cv.size()));
+  assert(cnt_ >= 0);
+  cv.clear();
+  cv.reserve(cv.size() + results.size());
```

Comment:
I think this can just be  `cv.reserve(results.size())` if you want

Reply:
Yep, updated.